File: app/src/main/java/ims/yang/com/ims/fragment/FriendListFragment.java
package ims.yang.com.ims.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import ims.yang.com.ims.R;
import ims.yang.com.ims.adapter.FriendAdapter;
import ims.yang.com.ims.entity.Friend;

import java.util.ArrayList;
import java.util.List;

/**
 * @author yangchen
 * on 3/12/2019 12:03 AM
 */
public class FriendListFragment extends Fragment {
    SwipeRefreshLayout swipeRefreshLayout;
    RecyclerView recyclerView;
    FriendAdapter friendAdapter;

    private List<Friend> getFriends() {
        List<Friend> userList = new ArrayList<>();
        for (int i = 0; i < 30; i++) {
            Friend user = new Friend("好友昵称" + i);
            userList.add(user);
        }
        return userList;
    }

    /**
     * 刷新页面
     */
    public void refresh() {
        List<Friend> userList = new ArrayList<>();
        for (int i = 0; i < 30; i++) {
            Friend user = new Friend("我的好友昵称" + i);
            userList.add(user);
        }
        friendAdapter.setFriendList(userList);
        recyclerView.setAdapter(friendAdapter);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.list_frag, container, false);
        recyclerView = view.findViewById(R.id.msg_recycler_view);
        swipeRefreshLayout = view.findViewById(R.id.swipe_refresh);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
        recyclerView.setLayoutManager(layoutManager);
        friendAdapter = new FriendAdapter(getFriends());
        //设置适配器
        recyclerView.setAdapter(friendAdapter);
        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                //模拟发送网络请求
                //停止刷新
                swipeRefreshLayout.setRefreshing(false);
                refresh();

            }
        });
        return view;
    }

}

File: app/src/main/java/ims/chat/ui/fragment/FriendListFragment.kt
package ims.chat.ui.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import ims.chat.R

import ims.chat.adapter.FriendAdapter
import ims.chat.entity.Friend

import kotlinx.android.synthetic.main.list_frag.*

import java.util.ArrayList

/**
 * @author yangchen
 * on 3/12/2019 12:03 AM
 */
class FriendListFragment : Fragment() {
    private lateinit var friendAdapter: FriendAdapter

    private val friends: List<Friend>
        get() {
            val userList = ArrayList<Friend>()
            for (i in 0..29) {
                val user = Friend("好友昵称$i")
                userList.add(user)
            }
            return userList
        }

    /**
     * 刷新页面
     */
    fun refresh() {
        val friendList = ArrayList<Friend>()
        for (i in 0..29) {
            val user = Friend("我的好友昵称$i")
            friendList.add(user)
        }
        friendAdapter.setFriendList(friendList)
        rcyViewMsg.adapter = friendAdapter
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val layoutManager = LinearLayoutManager(activity)
        rcyViewMsg.layoutManager = layoutManager
        friendAdapter = FriendAdapter(friends)
        //设置适配器
        rcyViewMsg.adapter = friendAdapter
        swipeToRefresh.setOnRefreshListener {
            //模拟发送网络请求
            //停止刷新
            swipeToRefresh.isRefreshing = false
            refresh()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.list_frag, container, false)
    }

}
